sql/item.cc

```diff
@@ -9387,13 +9387,11 @@ void Item_trigger_field::cleanup()
 
 Item_result item_cmp_type(Item_result a,Item_result b)
 {
-  if (a == STRING_RESULT && b == STRING_RESULT)
-    return STRING_RESULT;
-  if (a == INT_RESULT && b == INT_RESULT)
-    return INT_RESULT;
-  else if (a == ROW_RESULT || b == ROW_RESULT)
+  if (a == b)
+    return a;
+  if (a == ROW_RESULT || b == ROW_RESULT)
     return ROW_RESULT;
-  else if (a == TIME_RESULT || b == TIME_RESULT)
+  if (a == TIME_RESULT || b == TIME_RESULT)
     return TIME_RESULT;
```

Comment:
the above two check is not needed any more as it will return via (a==b) check.

Reply:
its an OR condition rather than an AND so its still needed. a == TIME_RESULT || b == STRING_RESULT needs to return TIME_RESULT like it previously did

Reply:
i see. misread the code

Reply:
No worries. I appreciate your efforts that make me verify that I've implemented this correctly.